clang/test/SemaSYCL/sycl-device-const-static.cpp

```diff
@@ -1,26 +1,51 @@
 // RUN: %clang_cc1 -fsycl -fsycl-is-device -verify -fsyntax-only %s
 
+struct Base {};
+struct S {
+  __attribute__((sycl_device)) void foo();
```

Comment:
It was needed because there was a call to it from kernel code, and it was undefined (just declared).  I have now removed the attribute and instead simply defined it in-class instead.